README.md

```diff
@@ -255,11 +338,69 @@ custom function for logging cache hits/misses. called every `logInterval` second
 Example  
 
 ```js
-  logReport (cacheReport) {
-    console.log(`Cache report - ${cacheReport}`)
+  logReport (report) {
+    console.log('Periodic cache report', report)
+  }
+
+  // report format
+  {
+    "Query.add": {
+      "dedupes": 0,
+      "hits": 8,
+      "misses": 1,
+      "skips": 0
+    },
+    "Query.sub": {
+      "dedupes": 0,
+      "hits": 2,
+      "misses": 6,
+      "skips": 0
+    },
   }
 ```
 
+## Invalidation
+
+Along with `time to live` invalidation of the cache entries, we can use invalidation by keys.  
+The concept behind invalidation by keys is that entries have an auxiliary key set that explicitly link requests along with their own result. These axiliary keys are called here `references`.  
+The use case is common. Let's say we have an entry _user_ `{id: 1, name: "Alice"}`, it may change often or rarely, the `ttl` system is not accurate:
+
+- it can be updated before `ttl` expiration, this case the old value is showed until expiration by `ttl`.  
+It may also be in different queries, for example `getUser` and `findUsers`, so we need to keep their responses consistent
+- it's not been updated during `ttl` expiration, so in this case we don't need to reload the value, because it's not changed
+
+To solve this common problem, we can use `references`.  
+We can say that the result of query `getUser(id: 1)` has reference `user~1`, and the result of query `findUsers`, containing `{id: 1, name: "Alice"},{id: 2, name: "Bob"}` has references `[user~1,user~2]`.
+So we can find the results in the cache by their `references`, independently of the request that generated them, and we can invalidate by `references`.
+
+When the mutation `updateUser` involves `user {id: 1}` we can remove all the entries in the cache that have references to `user~1`, so the result of `getUser(id: 1)` and `findUsers`, and they will be reloaded at the next request with the new data - but not the result of `getUser(id: 2)`.
+
+However, the operations required to do that could be expensive and not worthing it, for example is not recommendable to cache frequently updating data by queries of `find` that have pagination/filtering/sorting.
+
+Explicit invalidation is `disabled` by default, you have to enable in `storage` settings.
+
+See [mercurius-cache-example](https://github.com/simone-sanfratello/mercurius-cache-example) for a complete example.
+
+### Redis
+
+Using a `redis` storage is the best choice for a shared cache for a cluster of a service instance.  
+However, using the invalion system need to keep `references` updated, and remove the expired ones: while expired references does not compromise the cache integrity, they slow down the invalidation task.  
```

Comment:
It should be clearer that the system does not leak records if there is no gc strategy. However it would operate faster/better if one was added.